Bad error message when a repository definition contains a non-pipeline-like object

The error message produced when a pipeline contains a non-pipeline-like object is wrong, due to a couple of missing parens in the `_Repository` code:

```
        if isinstance(repository_definitions, list):
            bad_definitions = []
            for i, definition in enumerate(repository_definitions):
                if not (
                    isinstance(definition, PipelineDefinition)
                    or isinstance(definition, PartitionSetDefinition)
                    or isinstance(definition, ScheduleDefinition)
                ):
>                   bad_definitions.append(i, type(definition))
E                   TypeError: append() takes exactly one argument (2 given)
```

It's a simple fix so I'll push a PR with a test.

--- a/python_modules/dagster/dagster/core/definitions/decorators/repository.py
+++ b/python_modules/dagster/dagster/core/definitions/decorators/repository.py
@@ -41,7 +41,7 @@ class _Repository(object):
                     or isinstance(definition, PartitionSetDefinition)
                     or isinstance(definition, ScheduleDefinition)
                 ):
-                    bad_definitions.append(i, type(definition))
+                    bad_definitions.append((i, type(definition)))
             if bad_definitions:
                 raise DagsterInvalidDefinitionError(
                     'Bad return value from repository construction function: all elements of list '

--- a/python_modules/dagster/dagster_tests/core_tests/definitions_tests/test_repository_definition.py
+++ b/python_modules/dagster/dagster_tests/core_tests/definitions_tests/test_repository_definition.py
@@ -133,3 +133,11 @@ def test_key_mismatch():
 
     with pytest.raises(Exception, match='name in PipelineDefinition does not match'):
         some_repo.get_pipeline('foo')
+
+
+def test_non_pipeline_in_pipelines():
+    with pytest.raises(DagsterInvalidDefinitionError, match='all elements of list must be of type'):
+
+        @repository
+        def _some_repo():
+            return ['not-a-pipeline']

